### DISCUSSION TITLE
Collecting a sequence of keystrokes (and their RTs) in one trial

### DISCUSSION THREAD

**Discussion by CarolynBuck-Gengler** [jspsych/jsPsych#801] [Q&A] [answered]
Thank you for the workshop today! This is a fantastic tool.

To stretch myself, I wanted to try programming an experiment I did in our lab a few years back. It entails collecting multiple keystrokes in a trial, which is ended by the participant typing the enter key (e.g., four digits are displayed, and the RTs for  all four digits' keystrokes and the <enter> key are collected). I realized that jspsych-html-keyboard-response only collects a single keystroke, and didn't see another plugin that appeared to do what I'm trying to do. 

I found the paper by Pinet et al. (2017) on measuring sequences of keystrokes, and that seems to be similar to what I want to try; in it they used jsPsych and specifically the plugins jspsych-key-sequence and jspsych-multi-stim-multi-response, which do not appear to be in the current jsPsych.

Is this something that needs more experience with jsPsych and/or writing or modifying my own plugin, or am I just not seeing the plugin I need to use to do this?

Thank you!
Carolyn

**Comment by jodeleeuw** [comment, marked as answer]
Hi Carolyn,

Thanks for attending the workshop!

We removed the multi-stim-multi-response plugin a few versions ago because in 99% of cases it was simpler to use a combination of other plugins. For example, just using several html-keyboard-response plugins in sequence.

I believe jspsych-key-sequence is a custom plugin that Pinet et al. wrote for their task.

I think your task is probably doable with the html-keyboard-response plugin. You could set up four consecutive trials, each with the same stimulus. Each trial would record the RT for one of the digits. From the perspective of the participant, it will seem like one continuous trial.

**Reply by CarolynBuck-Gengler** [reply]
Thank you for your quick response! 
I came to the same conclusion, to string together 4 (actually 5, one for the enter key) html-keyboard-response trials, for the particular thing I described. However, that would force the participant to enter exactly four digits + enter (obviously the ideal, but not what all participants did), and it would not work for other experiments (not involving digits) in which a variable response is required -- how would one program it if one wanted word responses, for instance? And could one capture backspacing to correct the response? 
I am going to try the approach of stringing together several trials.
Thanks again!
C
